File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode slow=head;
        ListNode fast=head.next;
        while(true){
            if(fast==null){
                return slow;
            }
            else if(fast.next==null){
                return slow.next;
            }
            slow=slow.next;
            fast=fast.next.next;
        }
    }
}

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func middleNode(head *ListNode) *ListNode {
	slow, fast := head, head.Next

	for ; fast != nil && fast.Next != nil; slow, fast = slow.Next, fast.Next.Next {

	}
	
    if fast==nil{
        return slow
    }
    return slow.Next
}